assembl/static2/js/app/components/debate/common/postTranslate.jsx

```diff
@@ -1,35 +1,118 @@
 // @flow
 import React from 'react';
+import { connect } from 'react-redux';
 import { Translate, I18n } from 'react-redux-i18n';
+
+import { updateContentLocaleById, updateContentLocaleByOriginalLocale } from '../../../actions/contentLocaleActions';
 import SwitchButton from '../../common/switchButton';
+import { getConnectedUserId } from '../../../utils/globalFunctions';
+import { displayCustomModal } from '../../../utils/utilityManager';
+import CancelTranslationForm from './cancelTranslationForm';
+import ChooseContentLocaleForm from './chooseContentLocaleForm';
 
-const PostTranslate = ({
-  id,
-  originalBodyLocale,
-  showOriginal,
-  toggle
-}: {
+type PostTranslateProps = {
+  contentLocale: string,
   id: string,
-  originalBodyLocale: string,
-  showOriginal: boolean,
-  toggle: Function
-}) => {
-  return (
-    <div className="translate">
-      <p>
-        <Translate
-          value={!showOriginal ? 'debate.thread.messageTranslatedFrom' : 'debate.thread.messageOriginallyIn'}
-          language={I18n.t(`language.${originalBodyLocale}`)}
+  lang: string,
+  originalLocale: string,
+  translate: boolean,
+  updateById: (value: string) => void,
+  updateByOriginalLocale: (value: string) => void
+};
+
+type PostTranslateState = {
+  isTranslated: boolean
+};
+
+class PostTranslate extends React.Component<void, PostTranslateProps, PostTranslateState> {
+  props: PostTranslateProps;
+  state: PostTranslateState;
+
+  constructor() {
+    super();
+    this.state = {
+      isTranslated: false
+    };
+  }
+
+  handleSubmit = () => {
+    const { contentLocale, lang, originalLocale, updateById } = this.props;
+    const userIsConnected = getConnectedUserId();
+    if (!userIsConnected) {
+      if (contentLocale && contentLocale !== lang) {
+        updateById(lang);
+      } else {
+        updateById(originalLocale);
+      }
+    } else {
+      this.openModal();
+    }
+  };
+
+  openModal = () => {
+    const { id, lang, originalLocale, translate, updateById, updateByOriginalLocale } = this.props;
+    let content;
+    if (!translate) {
+      content = (
+        <ChooseContentLocaleForm
+          id={id}
+          lang={lang}
+          originalLocale={originalLocale}
+          translate={translate}
+          updateById={updateById}
+          updateByOriginalLocale={updateByOriginalLocale}
+        />
+      );
+    } else {
+      content = (
+        <CancelTranslationForm
+          lang={lang}
+          originalLocale={originalLocale}
+          translate={translate}
+          updateById={updateById}
+          updateByOriginalLocale={updateByOriginalLocale}
+        />
+      );
+    }
+
+    return displayCustomModal(content);
+  };
+
+  render() {
+    const { id, lang, originalLocale, translate } = this.props;
+    const specialLocaleCodes = ['und', 'zxx']; // locale codes that can not be translated
```

Comment:
What is the language associated to this locale?

Reply:
`und` is for `undefined` and `zxx` is another special locale when the system can't find what it is

Reply:
ok good to know. Can you add it in the comment?

Reply:
done